code-styling/mysql/README.md

```diff
@@ -0,0 +1,574 @@
+
+```sql
+SELECT column_a
+FROM my_table
+WHERE id = 5;
+
+SELECT
+  column_a,
+  column_b,
+  column_c
+FROM my_table
+WHERE id < 100
+AND created_at >= '2017-01-01';
+
+SELECT
+  column_a,
+  column_b,
+  column_c
+FROM my_table
+WHERE (
+  id < 100
+  AND created_at >= '2017-01-01'
+) OR (
+  id < 50
+  AND created_at > '2016-06-30 12:34:56'
+)
+```
+
+Bad select queries:
+
+```sql
+SELECT column_a
+FROM
+  my_table
+WHERE id = 5;
+
+SELECT column_a
+FROM my_table
+WHERE
+  id = 5;
+
+SELECT column_a, column_b, column_c
+FROM my_table
+WHERE id < 100 AND created_at >= '2017-01-01';
+
+-- Missing brackets
+SELECT
+  column_a,
+  column_b,
+  column_c
+FROM my_table
+WHERE id < 100
+AND created_at >= '2017-01-01'
+OR id < 50
+AND created_at > '2016-06-30 12:34:56'
+```
+
+### Joins
+
+- The `JOIN` keyword MUST NOT be used without specifying the join type
+
+- The preferred join type is `INNER JOIN`
+
+- `LEFT JOIN` SHOULD only be used if the results may contain null values
+  for the joined table
+
+- `RIGHT JOIN` SHOULD be avoided
+
+- The `ON` clause MUST be written on the next line with one indent level
+
+Good join queries:
+
+```sql
+SELECT
+  t1.id,
+  t1.column_a,
+  t2.column_b
+FROM table1 AS t1
+INNER JOIN table2 AS t2
+  ON t1.id = t2.table1_id;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id
+  AND t2.table1_id IS NULL;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id
+  AND t2.updated_at IS NULL;
+```
+
+Bad join queries:
+
+```sql
+SELECT
+  t1.id,
+  t1.column_a,
+  t2.column_b
+FROM table1 AS t1
+INNER JOIN table2 AS t2 ON t1.id = t2.table1_id;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+ON t1.id = t2.table1_id
+AND t2.table1_id IS NULL;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id AND t2.table1_id IS NULL;
+```
+
+### Delete queries
+
+Delete queries SHOULD not be used.
+
+If a delete query is required, the table from which the rows are deleted MUST be
+specified on the same line as the `DELETE FROM` keyword. Delete queries also
+MUST specify a WHERE clause.
+
+Good delete queries:
+
+```sql
+DELETE FROM my_table
+WHERE id = 5;
+```
+
+Bad delete queries:
+
+```sql
+-- Invalid: `FROM` must be on same line as `DELETE`
+DELETE
+FROM my_table
+WHERE id = 5;
+
+-- Invalid: table name must be on same line as `FROM`
+DELETE FROM
+my_table
+WHERE id = 5;
+
+-- Where clause should follow `WHERE` keyword
+DELETE FROM my_table
+WHERE
+  id = 5;
+
+-- Missing `WHERE` clause
+DELETE FROM my_table;
+```
+
+### Insert queries
+
+Insert queries MUST specify all columns in which data will be inserted. Omitted
+columns MUST have a default value. The `id` column MUST NOT be specified in
+insert queries unless those queries are used in some kind of data replication.
+
+If an insert query specifies more than 5 columns, the `SET` syntax is
```

Comment:
Typo `5` should be `five`<sup>[1]</sup>

<sup>[1]: In writing it is common to use words rather than digits for numbers under 10 (or at the beginning of a sentence).</sup>